p2p/sync.go

```diff
@@ -60,83 +59,86 @@ func (s *syncService) start(ctx context.Context) {
 		s.log.Debugw("Continuous iteration", "i", i)
 
 		iterCtx, cancelIteration := context.WithCancel(ctx)
-
-		var nextHeight int
-		if curHeight, err := s.blockchain.Height(); err == nil {
-			nextHeight = int(curHeight) + 1
-		} else if !errors.Is(err, db.ErrKeyNotFound) {
-			s.log.Errorw("Failed to get current height", "err", err)
+		nextHeight, err := s.getNextHeight()
+		if err != nil {
+			s.logError("Failed to get current height", err)
```

Comment:
I'd suggest we add block height data in the error log

Reply:
```suggestion
			s.logError("Failed to get current height ", fmt.Errorf("current Height: %d, err: %w", nextHeight-1, err)
```